src/VecSim/vec_sim_common.h

```diff
@@ -130,9 +136,46 @@ typedef struct {
     } specificParams;
 } TieredIndexParams;
 
+typedef struct {
+    VecSimType type;                   // Datatype to index.
+    size_t dim;                        // Vector's dimension.
+    VecSimMetric metric;               // Distance metric to use in the index.
+    bool multi;                        // Determines if the index should multi-index or not.
+    size_t nLists;                     // Number of inverted lists
+    bool conservativeMemoryAllocation; // Use as little GPU memory as possible
+    size_t kmeans_nIters;              // Iterations for kmeans calculation
+    double kmeans_trainsetFraction;    // Fraction of dataset used for kmeans training
+    unsigned nProbes;                  // The number of clusters to search
+    size_t pqBits;                     // Bit length of vector element after PQ compression. If set
+                                       // to 0, IVF flat will be used instead of IVFPQ.
+    // ***************** IVF-Flat-only parameters ******************
+    // The following parameters will be ignored if pqBits is set to a
+    // non-zero value.
+    bool adaptiveCenters; // If index should be updated for new vectors
+
+    // ******************* IVFPQ-only parameters *******************
+    // The following parameters will be ignored if pqBits is set to 0
+    size_t pqDim; // The dimensionality of an encoded vector after PQ
+                  // compression. If set to 0, a heuristic will be used to
+                  // select the dimensionality.
+
+    IVFPQCodebookKind codebookKind;
+    CudaType lutType;
+    CudaType internalDistanceType;
+    double preferredShmemCarveout; // Fraction of GPU's unified memory / L1
+                                   // cache to be used as shared memory
+
+} IVFParams;
+
+typedef struct {
```

Comment:
I don't believe we need this struct definition as is
you can send the `IVFParams` as `primaryIndexParams` when creating `TieredIndexParams`
if you need something specific for you tiered index management layer, this is the case where you need to create a specific struct 
see 
```
    union {
        TieredHNSWParams tieredHnswParams;
    } specificParams;
``` 
at `TieredIndexParams`